pilot/cmd/pilot-agent/status/server.go

```diff
@@ -204,8 +204,9 @@ func NewServer(config Options) (*Server, error) {
 		if config.IPv6 {
 			localAddr = upstreamLocalAddressIPv6
 		}
-		d := &net.Dialer{
-			LocalAddr: localAddr,
+		d := &net.Dialer{}
+		if config.StatusPort != 0 {
```

Comment:
I am not sure I understand what this is for?

Reply:
I assume `config.StatusPort` being set to 0 would mean that the server would bind to any port, but we should still have our local address be `127.0.0.6` or the ipv6 equivalent, right? Not sure how the two are related.

Reply:
Because this is CI failure https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/31981/unit-tests_istio/1380002378433957888